src/spec/create_extension_spec.py

```diff
@@ -178,6 +198,75 @@ def main():
         neurodata_type_def="NIRSDevice",
         neurodata_type_inc="Device",
         doc="A NIRS Device",
+        attributes=[
+            NWBAttributeSpec(
+                name="nirs_mode",
+                doc=(
+                    "The mode of NIRS measurement performed with this device (e.g."
+                    " continuous-wave, frequency-domain, time-domain,"
+                    " time-domain-moments, diffuse-correlation-spectroscopy,"
+                    " continuous-wave-fluorescence, etc)"
+                ),
+                dtype="text",
+            ),
+            NWBAttributeSpec(
+                name="frequency",
+                doc=(
+                    "The modulation frequency in Hz used for frequency domain NIRS."
+                    " if nirs_mode is a type of frequency domain spectroscopy."
+                ),
+                dtype="float",
+                required=False,
+            ),
+            NWBAttributeSpec(
+                name="time_delay",
+                doc=(
+                    "The time delay in ns used for gated time domain NIRS. Only used"
+                    " if nirs_mode is a type of gated time domain spectroscopy."
+                ),
+                dtype="float",
+                required=False,
+            ),
+            NWBAttributeSpec(
+                name="time_delay_width",
+                doc=(
+                    "The time delay width in ns used for gated time domain NIRS. Only"
+                    " used if nirs_mode is a type of gated time domain spectroscopy."
+                ),
+                dtype="float",
+                required=False,
+            ),
+            NWBAttributeSpec(
+                name="correlation_time_delay",
+                doc=(
+                    "The correlation time delay in ns for diffuse correlation"
+                    " spectroscopy NIRS. Only used if nirs_mode is a type of diffuse"
+                    " correlation spectroscopy."
+                ),
+                dtype="float",
+                required=False,
+            ),
+            NWBAttributeSpec(
+                name="correlation_time_delay_width",
+                doc=(
+                    "The correlation time delay width in ns for diffuse correlation"
+                    " spectroscopy NIRS. Only used if nirs_mode is a type of diffuse"
+                    " correlation spectroscopy."
+                ),
+                dtype="float",
+                required=False,
+            ),
```

Comment:
I see for these you defined the units as `ns`. Is that from the snirf spec? I wonder if there is/will be any case where other units will be more useful on any of these. That could be a good thing to ask a NIRS researcher. 
An option would be to put units in separate attributes, e.g. 
```
name = "time_delay_units"
doc = ("The units for time_delay, usually in `ns`")
dtype = "text"
required = False
```
but again, I think ns is good for now and we can wait until someone requests different units.

Reply:
nanoseconds was chosen because I looked up some experimental values from TD-NIRS research papers. But I agree, it'd be great to get input from a NRIS researcher, and also add the units as a separate attribute (see #7)